### ISSUE TITLE
3.1.2: Test failure on big endian architectures in test_coldefs_init_from_array

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#8458]
Running the tests during build time for version 3.1.2 shows the following failure in a newly added test:
```
_______________ TestColumnFunctions.test_coldefs_init_from_array _______________

self = <astropy.io.fits.tests.test_table.TestColumnFunctions object at 0x3ff7efbeb38>

    def test_coldefs_init_from_array(self):
        """Test that ColDefs._init_from_array works with single element data-
        types as well as multi-element data-types
        """
        nd_array = np.ndarray((1,), dtype=[('A', '<u4', (2,)), ('B', 'uint16')])
        col_defs = fits.column.ColDefs(nd_array)
>       assert 2**31 == col_defs['A'].bzero
E       AssertionError: assert (2 ** 31) == None
E        +  where None = name = 'A'; format = '2J'.bzero

astropy/io/fits/tests/test_table.py:2935: AssertionError
====== 1 failed, 11667 passed, 358 skipped, 66 xfailed in 130.01 seconds =======
```
This happens on all big endian architectures (32 and 64 bit): MIPS, IBM zSystems, PowerPC, HP-PA.
Full build log (for MIPS) [here](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=mips&ver=3.1.2-1&stamp=1551131794&raw=0).

**Comment by saimn** [comment]
Was added in #8353, cc @subagonsouth.
At first sight I think that the test is correct and it is the check in `_init_from_array` that does not correctly check for unsigned dtypes.

**Comment by subagonsouth** [comment]
I don't have a ton of numpy experience, but it appears that the checking should be looking for `ftype.base.kind == 'u'` for all 3 cases in the `if` statement. The fix that I submitted failed to account for datatypes of a different endianess from the host machine.

**Comment by mhvk** [comment]
@subagonsouth - if you can find time to look into this, that would be most welcome!  I assigned this issue to you for now, please do let us know if you do not have time, as getting a fix is important.

Which does bring up a question for @astrofrog and @bsipocz - is there any CI where we run our tests on a big-endian machine?

Or, @olebole, is there a way we can test things on Debian machines right before a release? (We probably do not have to check every commit, since this comes about rarely enough.)

**Comment by mhvk** [comment]
@subagonsouth - I don't seem to able to put you in the "assignees" section, but consider yourself assigned anyway!

**Comment by mhvk** [comment]
One possible option: http://osuosl.org/services/powerdev/ (see halfway down the short page)

**Comment by bsipocz** [comment]
> Which does bring up a question for @astrofrog and @bsipocz - is there any CI where we run our tests on a big-endian machine?

I'm not aware of that any such architecture is available, but if anyone knows about a configuration on any of the standard CI services we surely can have one job of the matrix to run on that.

**Comment by olebole** [comment]
@mhvk I have access to such a machine and could test things if you give me a patch. For a certain issue, you can also [request own access](https://dsa.debian.org/doc/guest-account/) -- I would of course sponsor such a request. This however would always take a week or so.
Usually I try to upload prereleases (rc and such) to our "experimental" branch if my time permits to check for potential problems. But I don't see a good workflow how to do this more often -- maybe I can find time to automate my work a bit so that it is just a button press (would be nice for other things as well, like compatibility with packages that depend on astropy), but my (unfunded) resources are unfortunately limited here.
For CI tests on big endian, the only idea I would have is setting up a qemu (slow, will take ages for all astropy tests).

**Comment by subagonsouth** [comment]
Pull request for fix has been submitted...
https://github.com/astropy/astropy/pull/8460

**Comment by MSeifert04** [comment]
Closed in #8460
